store/file/ods.go

```diff
@@ -108,22 +108,45 @@ func writeODSFile(w io.Writer, eds *rsmt2d.ExtendedDataSquare, axisRoots *share.
 		return nil, fmt.Errorf("writing axis roots: %w", err)
 	}
 
-	for _, shr := range eds.FlattenedODS() {
-		if _, err := w.Write(shr); err != nil {
-			return nil, fmt.Errorf("writing shares: %w", err)
-		}
+	// write quadrants
+	err = writeQuadrant(w, eds, 0)
+	if err != nil {
+		return nil, fmt.Errorf("writing Q1: %w", err)
 	}
 	return h, nil
 }
 
+func writeQuadrant(w io.Writer, eds *rsmt2d.ExtendedDataSquare, quadrantIdx int) error {
+	buf := make([]byte, 0, share.Size*eds.Width())
+	fromRow := quadrantIdx / 2 * int(eds.Width()) / 2
+	toRow := fromRow + int(eds.Width())/2
+
+	fromCol := quadrantIdx % 2 * int(eds.Width()) / 2
+	toCol := fromCol + int(eds.Width())/2
+
+	for rowIdx := fromRow; rowIdx < toRow; rowIdx++ {
+		row := eds.Row(uint(rowIdx))
+		for colIdx := fromCol; colIdx < toCol; colIdx++ {
+			buf = append(buf, row[colIdx]...)
+		}
+		if _, err := w.Write(buf); err != nil {
+			return fmt.Errorf("writing shares: %w", err)
+		}
+		buf = buf[:0]
+	}
```

Comment:
I don't follow. Why can't we just iterate over half of the rows and write every half of it? Why do we need columns involved at all?

Reply:
Implementation would be much easier with `bufio.Writer` tbh.

Reply:
We need both rows and columns to be able to write any quadrant, not only the first one.
I think basic slice with append is as simple as it can get (only 3 lines of code). Can replace those 3 lines with 3 lines of bufio.Writer, if you think it is easier to read